File: Blind75/Arrays/11. Container With Most Water.py
#11. Container With Most Water

class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        res,l,r=0,0,n-1
        while(l<r):
            area=(r-l)*min(height[l],height[r])
            res=max(res,area)
            if(height[l]<height[r]):
                l+=1
            else:
                r-=1
        return res

File: NeedCode 150 (JAVA)/Two Pointer/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int max_water = 0;
        int l = 0, r = height.length - 1;
        while (l < r) {
            int water = (r - l) * Math.min(height[l], height[r]);
            max_water = Math.max(max_water, water);
            if (height[l] < height[r]) {
                l++;
            } else
                r--;
        }
        return max_water;
    }
}
